src/server/pps/server/api_server.go

```diff
@@ -1041,6 +1041,9 @@ func (a *apiServer) validatePipeline(ctx context.Context, pipelineInfo *pps.Pipe
 			return fmt.Errorf("contradictory parallelism strategies: must set at " +
 				"most one of ParallelismSpec.Constant and ParallelismSpec.Coefficient")
 		}
+		if pipelineInfo.Service != nil && (pipelineInfo.ParallelismSpec.Constant > 1 || pipelineInfo.ParallelismSpec.Coefficient > 0) {
```

Comment:
Wouldn't it be easier to check `pipelineInfo.ParallelismSpec.Constant != 1`?

Reply:
Lol, good point.